packages/service/ni_measurement_plugin_sdk_service/discovery/_types.py

```diff
@@ -1,7 +1,14 @@
 """Data types for the NI Discovery Service."""
 
+from __future__ import annotations
```

Comment:
Why is this import required?

Reply:
In the below conversion method,

```
@classmethod
    def _from_grpc(cls, other: discovery_service_pb2.ServiceLocation) -> ServiceLocation:
         return ServiceLocation(
            location=other.location,
            insecure_port=other.insecure_port,
            ssl_authenticated_port=other.ssl_authenticated_port,
        )
```

the type hint `ServiceLocation` will throw error without `from __future__ import annotations`. 

Followed this from [packages/service/ni_measurement_plugin_sdk_service/session_management/_types.py](https://github.com/ni/measurement-plugin-python/blob/cccabe8cd06f11cfbc74fdd18feae1511cadc229/packages/service/ni_measurement_plugin_sdk_service/session_management/_types.py#L35).

Reply:
Okay, what is the error we see and do we know the reason why it throws that error?

Reply:
Without `from __future__ import annotations`, if you want to reference the class within the same class, we need to use string-based type hints.
Like,
```
@classmethod
    def _from_grpc(cls, other: discovery_service_pb2.ServiceLocation) -> "ServiceLocation":
         return ServiceLocation(
            location=other.location,
            insecure_port=other.insecure_port,
            ssl_authenticated_port=other.ssl_authenticated_port,
        )
```
if we use `ServiceLocation` type hint without string literal in the above case, we will face 

> NameError: name 'ServiceLocation' is not defined. 

**References :** [Link1](https://stackoverflow.com/questions/33533148/how-do-i-type-hint-a-method-with-the-type-of-the-enclosing-class), [Link2](https://realpython.com/python-type-self/)

From Python 3.7, we can use `from __future__ import annotations`, to use the class name without string literal. As this approach was used the current repo, I choosed the same to follow uniformity.